### PR TITLE
Initializes Home View front end + adds User.recentlyViewedRecipes functionality

### LINKED ISSUE #45: get nutrition stats functionality running (backend)

### PR DESCRIPTION
**Pull request by parkhangorodsky**
this branch: 
- initializes the home view
- changes the home view to be the first view shown upon sign in
- adds User.recentlyViewedRecipes functionality (though currently uses a placeholder in the HomeView until the other views are initialized after login)

what is NOT done in this branch (to be done in a future issue):
- complete HomeViewModel.
- add the bar charts for the nutrition info.
- adds NutritionStats use case (which is a changed copy of NutritionDisplay), since NutritionDisplay is now useless, and this use case will directly update the HomeView (this is done in another branch)

### CODE REVIEW

File: src/main/java/entity/User.java [hunk 1]
```diff
@@ -35,6 +35,8 @@ public User(String userName, String userEmail, String userPassword, LocalDateTim
         this.shoppingLists = new ArrayList<>();
         this.recipes = new ArrayList<>();
         this.preference = new HashMap<>();
+        this.recentlyViewedRecipes = new ArrayList<>();
+        this.recentlyViewedRecipes.add(new Recipe());
```
Comment (seonghyunban) [inline src/main/java/entity/User.java]: what is this?

File: src/main/java/entity/User.java [hunk 2]
```diff
@@ -61,6 +64,15 @@ public User() {}
     public void setRecipes(List<Recipe> recipes) {this.recipes = recipes;}
     public void setShoppingLists(List<ShoppingList> shoppingLists) {this.shoppingLists = shoppingLists;}
     public void setPreference(Map<String, Object> preference) {this.preference = preference;}
+    public void addRecentlyViewedRecipe(Recipe recipe) {
+        if (this.recentlyViewedRecipes.size() >= 5) {
+            this.recentlyViewedRecipes.remove(0); // one sec
```
Comment (seonghyunban) [inline src/main/java/entity/User.java]: Yep

File: src/main/java/entity/User.java [hunk 3]
```diff
@@ -61,6 +64,15 @@ public User() {}
...
+            this.recentlyViewedRecipes.add(recipe);
+        }
+        else {
+            this.recentlyViewedRecipes.add(recipe);
```
Comment (seonghyunban) [inline src/main/java/entity/User.java]: if (this.recentlyViewedRecipes.size() >= 5) {
            this.recentlyViewedRecipes.remove(0);
}
this.recentlyViewedRecipes.add(recipe);

File: src/main/java/use_cases/_common/gui_common/view/HomeView.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        if (evt.getPropertyName().equals("nightMode")) {
+            toggleNightMode();
+            this.revalidate();
+            this.repaint();
+        }
+        else if (evt.getPropertyName().equals("themeColor")) {
```
Comment (seonghyunban) [inline src/main/java/use_cases/_common/gui_common/view/HomeView.java]: What's this

File: src/main/java/use_cases/search_recipe/gui/view_component/RecipePanel.java [hunk 5]
```diff
@@ -76,6 +77,7 @@ public RecipePanel(Recipe recipe, DisplayRecipeDetailController displayRecipeDet
 
         detailButton = new RoundButton("Detail");
         detailButton.addActionListener(e -> {
+            LoggedUserData.getLoggedInUser().addRecentlyViewedRecipe(recipe);
```
Comment (seonghyunban) [inline src/main/java/use_cases/search_recipe/gui/view_component/RecipePanel.java:80]: You need to update this in the database

**Review by seonghyunban — CHANGES_REQUESTED** [review]
Please check the comments

File: src/main/java/use_cases/_common/gui_common/view/HomeView.java (hunk 4 above)
Comment (parkhangorodsky) [reply src/main/java/use_cases/_common/gui_common/view/HomeView.java]: removed

File: src/main/java/entity/User.java (hunk 3 above)
Comment (parkhangorodsky) [reply src/main/java/entity/User.java]: changed to this

File: src/main/java/entity/User.java (hunk 1 above)
Comment (parkhangorodsky) [reply src/main/java/entity/User.java]: removed line 39

File: src/main/java/use_cases/_common/gui_common/view/HomeView.java (hunk 4 above)
Comment (parkhangorodsky) [reply src/main/java/use_cases/_common/gui_common/view/HomeView.java:117]: to be addressed in the next PR/issue

File: src/main/java/use_cases/search_recipe/gui/view_component/RecipePanel.java (hunk 5 above)
Comment (parkhangorodsky) [reply src/main/java/use_cases/search_recipe/gui/view_component/RecipePanel.java:80]: to be addressed in the next PR/issue

**Review by seonghyunban — APPROVED** [review]
checked!

**Pull request merged**